matRad_calcLateralParticleCutOff.m

```diff
@@ -57,10 +54,29 @@
 % define function handles for single and double gauss
 SG =  @(vR,Sigma)((1/(2*pi*Sigma^2)).*exp(-(vR.^2)./(2*Sigma^2)));
 DG =  @(vR,Z,w,Sigma1,Sigma2) Z*(((1-w)*SG(vR,Sigma1)) + (w*SG(vR,Sigma2)));
-                         
-% loop over all entries in the machine.data struct
-for energyIx = 1:length(machine.data)
 
+% find for each energy the biggest foci index
+FociEnergyLUT  = unique([[stf.ray(:).focusIx] ; [stf.ray(:).energy]]','rows');
+UniqueEnergies = unique(FociEnergyLUT(:,2));
+N=histc(FociEnergyLUT(:,2),UniqueEnergies);
+FociEnergyLUT(N>1,:) = [];
```

Comment:
Something is fishy here. Please revise and also make sure that you are not dealing with a worst case estimation for the SSD,energy,initFokus combination!